src/contract.rs

```diff
@@ -0,0 +1,828 @@
+        // if end_block_height >= deal.end_block as u64 {
+        //      return Err(ContractError::DealClosedForBidding{});
+        // }
+        if deal.bid_token_denom!=msg.denom{
+            return Err(ContractError::DenomNotMatched{})
+        }
+        if msg.price < deal.min_price {
+            return Err(ContractError::MinimumPriceNotSatisfied{})
+        }
+        //retriveal of bidstore of respective deal_id
+        let mut bid_store = DEALSTORE.may_load(deps.storage,msg.deal_id)?;
+        //Generating the unique bid_id
+        let bid_id = BID_SEQ.update::<_,StdError>(deps.storage, |bid_id| Ok(bid_id.add(1)))?;
+        deal.total_bid=deal.total_bid+msg.amount;
+        let _res=DEALS.save(deps.storage, msg.deal_id, &deal);
+
+        //pushing the bids into the bidstore
+        if let Some(mut store) = bid_store {
+            store.bids.push((bid_id,bid));
+            DEALSTORE.save(deps.storage,msg.deal_id, &store)?;
+        } else {
+            let mut new_store = BidStore { bids: vec![]};
+            //creating new bidstore if not existed 
+            new_store.bids.push((bid_id,bid));
+            DEALSTORE.save(deps.storage,msg.deal_id, &new_store)?;
+        }
+        //to know the updated bid_store_list
+        let bid_store_updated= DEALSTORE.may_load(deps.storage,msg.deal_id)?;
+
+        // msg for locking bid deposit in the contract
+        let lock_funds_msg: BankMsg = BankMsg::Send{
+        to_address: _env.contract.address.to_string(),
+        amount: vec![Coin {
+            denom: msg.denom,
+            amount: msg.amount,
+        }]};
+        let messages = vec![lock_funds_msg];
+        Ok(Response::new().add_messages(messages).add_attribute("action", "place_bid").add_attribute("bid_id",bid_id.to_string()))
+}
+
+
+pub fn withdraw_bid(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    msg: WithdrawBidMsg,
+) -> Result<Response,ContractError> {
+
+    let res=DEALS.has(deps.storage,msg.deal_id);
+    if res==false{
+        return Err(ContractError::DealNotExisted{})
+    }
+    let mut deal=  DEALS.load(deps.storage,msg.deal_id)?;
+    let end_block_height=Uint128::from(_env.block.height);
+    // if end_block_height >= deal.end_block {
+    //      return Err(ContractError::CannotWithdrawBid{});
+    // }
+    // Read DEALSTORE from storage
+    let mut deal_store = DEALSTORE.load(deps.storage, msg.deal_id)?;
+    // Check if the bid_id is present in the bid_store
+    if let Some(index) = deal_store.bids.iter().position(|(bid_id, _)| *bid_id == msg.bid_id) {
+        
+        
+        let mut bank_msgs=Vec::new();
+        for (index, bid) in deal_store.bids.iter().enumerate() {
+            let bid_amount = bid.1.amount;
+            let denom = bid.1.denom.clone();
+            let bidder = bid.1.bidder.clone();
+            if info.sender.as_str()!=bidder {
+                return Err(ContractError::InvalidBidder {});
+            }; 
```

Comment:
Yes ,it will throw an error.We have modified the code making sure only retrieves data of respective deal_id and bid_id and checking whether signer is matching with bidder 
  if let Some(index) = deal_store.bids.iter().position(|(bid_id, _)| *bid_id == msg.bid_id) {
            let mut bank_msgs = Vec::new();
            let bid_amount = deal_store.bids[index].1.amount;
            let denom = deal_store.bids[index].1.denom.clone();
            let bidder =  deal_store.bids[index].1.bidder.clone();
            //checking whether the signer is matching with the bidder as per deal_id and bid_id
            if info.sender.as_str() != bidder {
                return Err(ContractError::InvalidBidder {});
            }